pyvrp/cpp/educate/LocalSearch.cpp

```diff
@@ -35,15 +39,25 @@ Individual LocalSearch::search(Individual &individual,
         for (auto const uClient : orderNodes)
         {
             auto *U = &clients[uClient];
+
             auto const lastTestedNode = lastTestedNodes[uClient];
             lastTestedNodes[uClient] = numMoves;
 
+            if (U->route && !data.client(uClient).required)  // test removing U
+                maybeRemove(U, costEvaluator);
+
             // Shuffling the neighbours in this loop should not matter much as
             // we are already randomizing the nodes U.
             for (auto const vClient : neighbours[uClient])
```

Comment:
In another PR, we should probably incorporate prizes in the granular neighborhood. I recall you mentioned this earlier somewhere but can't find it any longer. Should we make an issue of this?

More general: to what extent are you planning to add features to improve the performance on PC instances? Or is this enough?

Reply:
I'm going to make an issue collecting some of the things we discussed here. For prize collecting, my main improvement will through the neighbourhoods (for PyVRP I'm aiming for something simple, but also for another unrelated paper I'll also work on something like this). I'll try to also benchmark against Kleopatra's implementation, but that will take a few more days (and need not be part of this PR).

So basically only the neighbourhoods, unless we find that Kleopatra's implementation is significantly better than this simple thing.